src/Confluent.Kafka/Producer.cs

```diff
@@ -548,12 +613,35 @@ public void HandleDeliveryReport(Message message)
             }
         }
 
-        public void ProduceAsync(string topic, TKey key, TValue val, IDeliveryHandler<TKey, TValue> deliveryHandler, DateTime? timestamp = null, int? partition = null, bool blockIfQueueFull = true)
+        private void Produce(string topic, TKey key, TValue val, IDeliveryHandler<TKey, TValue> deliveryHandler, DateTime? timestamp, int partition, bool blockIfQueueFull)
         {
             var handler = new TypedDeliveryHandlerShim(key, val, deliveryHandler);
-            producer.ProduceAsync(topic, KeySerializer?.Serialize(key), ValueSerializer?.Serialize(val), handler, timestamp, partition, blockIfQueueFull);
+            var keyBytes = KeySerializer?.Serialize(key);
```

Comment:
Sorry for confusion, please see my comment https://github.com/confluentinc/confluent-kafka-dotnet/commit/10ee94437b13e9e5057966345737941f3235a3b3#commitcomment-20891497.